packages/core/http/core-http-server-internal/src/http_config.ts

```diff
@@ -385,8 +385,8 @@ export class HttpConfig implements IHttpConfig {
     this.requestId = rawHttpConfig.requestId;
     this.shutdownTimeout = rawHttpConfig.shutdownTimeout;
 
-    // default to `false` to prevent breaking changes in current offerings
-    this.restrictInternalApis = rawHttpConfig.restrictInternalApis ?? false;
+    // default to `true` to restrict internal APIs if not set through config.
+    this.restrictInternalApis = rawHttpConfig.restrictInternalApis ?? true;
```

Comment:
The actual breaking change.

Reply:
Should we rather change the defaults specified in config schema in this file?

https://github.com/elastic/kibana/blob/b2d6a5e0db4f27b4d9512d37be7a24ec032e10e5/packages/core/http/core-http-server-internal/src/http_config.ts#L210

We should also be able to delete the nullish check `?? ...` since `restrictInternalApis` will always have a value at this point.

Reply:
> Should we rather change the defaults specified in config schema in this file otherwise I'm not sure this work as intended.

Yea, we should.

Reply:
unfortunately, changing the default means updating a lot of tests...